aiida/common/log.py

```diff
@@ -161,7 +162,7 @@ def configure_logging(with_orm=False, daemon=False, daemon_log_file=None):
     """
     from logging.config import dictConfig
 
-    config = evaluate_logging_configuration(LOGGING)
+    config = evaluate_logging_configuration(copy.deepcopy(LOGGING))
```

Comment:
The name is correct in a sense that the `LOGGING` dictionary contains lambdas that need to be evaluated lazily. This is done by `evaluate_logging_configuration`. In that function I recursively iterate the mapping, creating copies of nested dictionaries exactly because the original should not be touched. However, it does not deal with sequences, so the nested lists in the evaluated copy will be references to the original. This leads to problems. I could have tried to add a clause for `isinstance(value, collections.Sequence)` but passing a deepcopy of the original structure is bullet-proof

Reply:
Ah ok - I thought "evaluate" was meant more in the sense of "analyze" than "execute".